File: benchmark/visualbasic/task8_http.vb
Imports System.Net.Http

Module HttpFetch
    Async Function FetchUrl(url As String) As Task(Of (Success As Boolean, Body As String))
        Try
            Using client As New HttpClient()
                Dim response = Await client.GetStringAsync(url)
                Return (True, response)
            End Using
        Catch ex As Exception
            Return (False, ex.Message)
        End Try
    End Function
End Module

File: benchmark/delphi/task8_http.pas
uses System.Net.HttpClient;

type
  TFetchResult = record
    Success: Boolean;
    Body: string;
  end;

function FetchUrl(const URL: string): TFetchResult;
var
  Client: THTTPClient;
  Response: IHTTPResponse;
begin
  Client := THTTPClient.Create;
  try
    try
      Response := Client.Get(URL);
      Result.Success := True;
      Result.Body := Response.ContentAsString;
    except
      on E: Exception do
      begin
        Result.Success := False;
        Result.Body := E.Message;
      end;
    end;
  finally
    Client.Free;
  end;
end;
